### ISSUE TITLE
Some styles issues

### ISSUE THREAD

**Issue by Calak** [pytesNET/tail.select#51]
Maybe these issues has been fixed in your next,  version, but just in case:

## 1 &ndash; Flickering outline on focus (Google Chrome)

**1.1** &mdash; With both "Default" and "Default Black" styles, when you click to open the tail.select, you got a blue outline flickering on Google Chrome.

![Style: Default](https://i.ibb.co/m4151fV/style-default.gif)

![Style: Default Black](https://i.ibb.co/0QzX1PS/style-default-black.gif)

**1.2** &mdash; With "Modern" style, when you got an unwanted blue rectangle once focused on it.

![Style: Modern](https://i.ibb.co/Rj416Ym/style-modern.gif)

To fix both, I simply added this to my css:

```
.tail-select {
    outline: none;
}
```

## 2 &ndash; Too long labels 

**2.1** &mdash; The Label's box model is "broken", you don't take the arrow's width into account and don't clip too long values.

**2.2** &mdash; Labels that are too long collide with the arrow, continues to the right, no matter what... Unless it contains spaces, in which case, it warp in multi lines (but the too long words continues to exeed the field)

One solution *could* be the following (That work with default styles but need some adjustments for bootstrap's ones.):

```
.tail-select .select-label {
    padding-right: 34px;  /* the .select-label::after width */
}
.tail-select .label-inner {
    display: block;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;    
}
.tail-select .label-count {
    float: left;
}
```

## 3 &ndash; User's selection

When double clicking on the tail.select, it select text of label or options. It's disturbing since it's no the result attended.

Simply adding the following fix that:

```
.tail-select * {
    user-select: none;
}
```

**Note**: for those fixes, you should also add vendors-specific prefixes

**Comment by SamBrishes** [comment]
Hellow,

thanks for your report and your respective fixes. Great Help! I'll keep this in mind, when I'm working on the themes / Less environment.

Thanks.

Sincerely,
Sam.

**Comment by SamBrishes** [comment]
Hellow, 

Short update here: I'm currently rewriting the Less environment + building a node.js script to auto-compile Less to CSS with all available color schemes (including minified stylesheets and sourcemaps for both versions). 

May take a while on 5 designs.

Sincerely,
Sam.

PS.: The new Less environment may looks like this (I'm currently playing around with the (newest) possibilities and features of the latest Less version):

```less
@import "tail.mixins.less"; // Contains all mixins for all Less Stylesheets

//
//  SETTINGS :: ALPHA COLOR
//
@alpha: transparent;

//
//  SETTINGS :: FONT FAMILY
//
@family: "Helvetica Neue", Helvetica, Arial, sans-serif;

//
//  SETTINGS :: USED COLOR SCHEME
//  The available settings are:
//      primary         Primary Blue
//      danger          Danger Red
//      warning         Warning Orange
//      success         Success Green
//      info            Info Blue (Light Blue)
//      invert          Invert Black
//
@color: primary;

//
//  SETTINGS :: COLOR SCHEMES
//   The @alpha are currently for testing the structure anbnd
//   gets replaced by the colors of course!
//
#scheme() when (@color = primary){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
#scheme() when (@color = danger){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
#scheme() when (@color = warning){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
#scheme() when (@color = success){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
#scheme() when (@color = info){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
#scheme() when (@color = invert){
    primary: @alpha;
    secondary: @alpha;
    foreground: @alpha;
    background: @alpha;

    .mark(){
        foreground: @alpha;
        background: @alpha;
    }
}
```

**Comment by Calak** [comment]
Thank you for telling me, I was going to propose a PR this weekend. Do you plan to release this script with version 0.5.13 (in this case I'll wait) or with version 1.0.0 (and therefore, I'll do a PR)

**Comment by SamBrishes** [comment]
Yes, the new styles / Less structure as well as the small node.js script will be released in 0.5.13 too.

**Comment by SamBrishes** [comment]
Short question @ the new stylings / structure

There are 2 (main) possibilities to manage the different color schemes:

1. Each color scheme has its own file, which would result into the image below. Many files, a bit confusing and a wall of CSS. 

The current solution is similar, except that the main stylings are stored in one file and just the different color properties / settings are stored separately (so currently you need to include 2 files to get the right design and color).

![bootstrap2](https://user-images.githubusercontent.com/31392477/60720029-baebc580-9f29-11e9-996e-980d907bde99.PNG)

2. Separate them with class names, which would reduce the number of files per style to 4 (.css, .map, .min, .min.map), but requires to set a specific class to the respective `tail.select()` constructor. Which may is (more) confusing and a breaking change?

Any other ideas?

Sincerely,
Sam.

**Comment by Calak** [comment]
Maybe implementing both alternatives for version 0.5.13 (the first, to keep retrocompability and the second to allow user to sets different scheme on different selects, or change scheme on-the-fly invalid selects) and for version 1 only keep the second alternative .

**Comment by SamBrishes** [comment]
I guess that's the best and easiest solution for the moment. Version 1.0.0 will also come with some cool new theming stuff, so it will be a little different anyway in the future.

---------

## Development Status
The node.js script is already finish.

#### Bootstrap 2
```
[========= ] 90%
```
+ Added two new color schemes: "funky" (pink) and "mystery" (purple), fetched from Bootstrap2

#### Bootstrap 3
```
[          ]  0%
```

#### Bootstrap 4
```
[          ]  0%
```

#### Default
```
[          ]  0%
```

#### Modern
```
[          ]  0%
```

**Comment by Calak** [comment]
For info, I'll wait the new version to make new PRs

**Comment by SamBrishes** [comment]
Okay, but keep in mind, that I want to release version 1.0.0, as soon as possible. 

I'm planning to use another 0.5.x updates only with urgent bugfixes.

**Comment by SamBrishes** [comment]
Hellow,

This has been fixed / added in Version 0.5.13. Thanks.

Sincerely,
Sam.
